pandas/io/common.py

```diff
@@ -274,14 +274,15 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
             import boto
         except:
             raise ImportError("boto is required to handle s3 files")
-        # Assuming AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+        # Assuming AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and AWS_S3_HOST
```

Comment:
I agree, it would be nice to have S3 documented in a dedicated section. Can that be its own GH issue so that this feature/bug can be pushed through faster?

I am not very familiar with the Pandas doc-string/documentation strategy for the read_* functions, so this may be something for a core contributor to work on.